crates/control/src/localization.rs

```diff
@@ -208,26 +208,44 @@ impl Localization {
                     None => {}
                 };
             }
-            (PrimaryState::Penalized, _, _)
-                if primary_state != PrimaryState::Penalized
-                    && (self.time_when_penalized_clicked.map_or(true, |time| {
+            (PrimaryState::Penalized, _, _) if primary_state != PrimaryState::Penalized => {
+                if self.was_picked_up_while_penalized
+                    || self.time_when_penalized_clicked.map_or(true, |time| {
                         context
                             .cycle_time
                             .start_time
                             .duration_since(time)
                             .expect("time ran backwards")
                             > *context.tentative_penalized_duration
-                    }) || !context.has_ground_contact) =>
-            {
-                if self.is_penalized_with_motion_in_set_or_initial {
-                    if self.was_picked_up_while_penalized_with_motion_in_set_or_initial {
-                        self.hypotheses = take(&mut self.hypotheses_when_entered_playing);
-
+                    })
+                {
+                    if self.is_penalized_with_motion_in_set_or_initial {
+                        if self.was_picked_up_while_penalized {
+                            self.hypotheses = take(&mut self.hypotheses_when_entered_playing);
```

Comment:
I think we should do something more intelligent: sometimes, the robot is picked up by accident during `is_penalized_with_motion_in_set_or_initial` and is placed at a similar location as before. We could keep the old hypothesis, reduce their probability and add the default penalized poses

Reply:
I agree, though this should be done in a separate PR. Feel free to open an issue for that.